### PR TITLE
Improve logging of background batch producer

### PR DESCRIPTION
**Pull request by aviflax**
@brunodecarvalho or @bdotdub please :eyes: thanks!

### CODE REVIEW

File: batchproducer/batchproducer.go [hunk 1]
```diff
@@ -261,17 +261,20 @@ func (b *batchProducer) sendBatch() {
 		b.currentStat.KinesisErrorsSinceLastStat++
 		b.logger.Printf("Error occurred when sending PutRecords request to Kinesis stream %v: %v", b.streamName, err)
 		b.returnRecordsToBuffer(records)
-	} else {
-		b.consecutiveErrors = 0
-		b.currentDelay = 0
-		succeeded := len(records) - res.FailedRecordCount
+		return
```
Comment (bdotdub) [inline batchproducer/batchproducer.go:264]: :+1: i dig the short circuit

File: batchproducer/batchproducer.go [hunk 2]
```diff
@@ -321,8 +324,8 @@ func (b *batchProducer) returnSomeFailedRecordsToBuffer(res *kinesis.PutRecordsR
 				b.records <- record
 			} else {
 				b.currentStat.RecordsDroppedSinceLastStat++
-				msg := "NOT re-enqueueing failed record for retry, as it has hit %v attempts, " +
-					"which is the maximum. Error code was: '%v' and message was '%v'"
+				msg := "Dropping failed record; it has hit %v attempts " +
+					"which is the maximum. Error code was: '%v' and message was '%v'."
```
Comment (bdotdub) [inline batchproducer/batchproducer.go:328]: :+1:

**Comment by bdotdub** [comment]
lgtm - this is mostly adding the success case, correct?

**Comment by aviflax** [comment]
> lgtm

great, thanks!

> this is mostly adding the success case, correct?

yes, and the logging, and the minor readability improvement to `sendBatch`.

**Comment by aviflax** [comment]
> yes, and the logging,

I meant "and the testing"

**Pull request merged**